### PR TITLE
Feature/skip manual address

### PR DESCRIPTION
**Pull request by dwybourn**
- When selecting an address from the dropdown, the user is no longer shown the manual address page.
- Fixed acceptance running with the address lookup feature enabled.

### CODE REVIEW

File: src/test/common/steps/address-lookup-steps.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ const { enterPostcodeAndSubmit } = require('./common-steps')
 const { setupPostcodeLookupWithNoResults, setupPostcodeLookupWithResults } = require('../wiremock')
 const { POSTCODE } = require('./constants')
 
-const POSTCODE_WITH_NO_RESULTS = 'BS1 1AA'
+const POSTCODE_WITH_NO_RESULTS = 'BS11AA'
```
Comment (dwybourn) [inline src/test/common/steps/address-lookup-steps.js]: Previous normalisation PR caused these tests to fail as the postcode sent (BS11AA) no longer matched that of the wiremock stub (BSA 1AA)

File: src/web/routes/application/select-address/select-address.js [hunk 2]
```diff
@@ -37,11 +39,14 @@ const findAddress = (selectedAddress, postcodeLookupResults) => {
 }
 
 const behaviourForPost = () => (req, res, next) => {
-  const address = findAddress(req.body.selectedAddress, req.session.postcodeLookupResults)
-  req.session.claim = {
-    ...req.session.claim,
-    ...transformAddress(address)
+  if (req.body.selectedAddress) {
```
Comment (dwybourn) [inline src/web/routes/application/select-address/select-address.js]: Bugfix, only transform the address if one has been selected. Can move to own pr if needed.

File: package.json [hunk 3]
```diff
@@ -10,21 +10,21 @@
...
-    "test:compatibility:local:debug": "mkdir -p build/reports/compatibility && cross-env SESSION_DETAILS_PORT=8081 OS_PLACES_URI=http://localhost:8090 TESTS=localCompatibility node --inspect-brk=9229 $NODE_DEBUG_OPTION node_modules/cucumber/bin/cucumber-js src/test/compatibility/features/ --require 'src/test/common/steps/**/*.js' --require 'src/test/common/timeout.js' --require 'src/test/compatibility/config/environment.js' -f json:build/reports/compatibility-report.json",
+    "test:compatibility:local:test": "mkdir -p build/reports/compatibility && cross-env SESSION_DETAILS_PORT=8081 OS_PLACES_URI=http://localhost:8090 OS_PLACES_URI=http://localhost:8090 TESTS=localCompatibility cucumber-js src/test/compatibility/features/ --require 'src/test/common/steps/**/*.js' --require 'src/test/common/timeout.js' --require 'src/test/compatibility/config/environment.js' -f json:build/reports/compatibility-report.json",
```
Comment (YetAnotherMatt) [inline package.json]: Duplicate OS_PLACES_URI in test:compatibility:local:test, test:acceptance:debug and test:compatibility:local:debug

File: src/web/routes/application/manual-address/manual-address.js [hunk 4]
```diff
@@ -32,13 +32,18 @@ const contentSummary = (req) => ({
   ])
 })
 
+const isNavigable = (session) => {
+  return isNil(session.claim.selectedAddress)
```
Comment (YetAnotherMatt) [inline src/web/routes/application/manual-address/manual-address.js]: I'm not sure this is going to work. What if someone selects an address in the dropdown, then decides that it's not the correct address and clicks 'my address is not listed'?

Comment (dwybourn) [reply src/web/routes/application/manual-address/manual-address.js]: This is handled in the behaviourForGet

File: package.json (hunk 3 above)
Comment (dwybourn) [reply package.json]: Removed duplicates.

File: src/web/routes/application/select-address/select-address.js [hunk 5]
```diff
@@ -20,6 +20,8 @@ const buildAddressOption = result => ({
 })
 
 const behaviourForGet = () => (req, res, next) => {
+  // reset this value as this is used to determine whether or not to skip the manual address page
+  delete req.session.claim.selectedAddress
```
Comment (mikeSmith08) [inline src/web/routes/application/select-address/select-address.js]: I'm not sure I understand why this is needed, lets discuss when you get in.

File: src/web/routes/application/select-address/select-address.test.js [hunk 6]
```diff
@@ -159,7 +200,7 @@ test('findAddress throws an error when the address is not found', (t) => {
 
   t.throws(
     findAddress.bind(null, selectedAddress, postcodeLookupResults),
-    'Unable to find selected address in list of postcode lookup results',
+    `Unable to find ${selectedAddress} in list of addresses: ${postcodeLookupResults}`,
```
Comment (gethyn1) [inline src/web/routes/application/select-address/select-address.test.js]: This test is incorrect as this is not the error that is thrown. I think this test is passing because this argument should be a regex

File: src/web/routes/application/select-address/select-address.js (hunk 5 above)
Comment (gethyn1) [reply src/web/routes/application/select-address/select-address.js]: Might be worth abstracting this to a named function to clarify the purpose. Something like:
```
const resetAddressState = (req) => {
  if (req.session.claim.selectedAddress) {
    delete req.session.claim.selectedAddress
  }
}
```

Comment (gethyn1) [reply src/web/routes/application/select-address/select-address.js]: Which could also be unit tested

File: src/web/routes/application/select-address/select-address.test.js (hunk 6 above)
Comment (dwybourn) [reply src/web/routes/application/select-address/select-address.test.js]: Updated to use a regex.

File: src/web/routes/application/select-address/select-address.js (hunk 5 above)
Comment (dwybourn) [reply src/web/routes/application/select-address/select-address.js]: Moved into own function and updated behaviourForGet unit test.

File: src/web/routes/application/select-address/select-address.test.js [hunk 7]
```diff
@@ -159,7 +203,7 @@ test('findAddress throws an error when the address is not found', (t) => {
 
   t.throws(
     findAddress.bind(null, selectedAddress, postcodeLookupResults),
-    'Unable to find selected address in list of postcode lookup results',
+    /Unable to find selected address in list of postcode lookup results/,
```
Comment (mikeSmith08) [inline src/web/routes/application/select-address/select-address.test.js]: Minor - shouldn't this (and the next line) be indented?
